doc/admin-guide/plugins/filter_body.en.rst

```diff
@@ -0,0 +1,440 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. include:: ../../common.defs
+
+.. _admin-plugins-filter_body:
+
+Filter Body Plugin
+******************
+
+Description
+===========
+
+The ``filter_body`` plugin provides streaming request and response body content
+inspection with configurable pattern matching and actions. It can be used to
+detect and mitigate security threats such as CVE exploits, XXE (XML External
+Entity) attacks, SQL injection patterns, and other malicious content.
+
+The plugin uses a streaming transform approach with a lookback buffer to handle
+patterns that may span buffer boundaries, avoiding the need to buffer the entire
+request or response body.
+
+Features
+--------
+
+- YAML-based configuration with flexible rule definitions.
+- Header-based filtering with AND/OR logic.
+- Case-insensitive header matching, case-sensitive body patterns.
+- Configurable actions per rule: ``log``, ``block``, ``add_header``.
+- Support for both request and response body inspection.
+- Streaming transform with lookback buffer for cross-boundary pattern matching.
+- Optional ``max_content_length`` to skip inspection of large bodies.
+- Configurable HTTP methods to match (GET, POST, PUT, etc.).
+- Optional ``status`` codes to match for response rules.
+- Per-rule metrics counters for monitoring match activity.
+
+Installation
+============
+
+The ``filter_body`` plugin is an experimental plugin. To build it, either pass
+``-DENABLE_FILTER_BODY=ON`` to ``cmake`` when configuring the build::
```

Comment:
Does experiemental plugins need to be turned on for this to work or will this flag turn on just this experimental plugin